### PR TITLE
[WIP] prevent timeout message loss when not using DTC

### PR DESCRIPTION
**Pull request by timbussmann**

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Timeouts/TransportWithFakeQueues.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        protected override void Configure(BusConfiguration config)
+        {
+            config.RegisterComponents(c => c.ConfigureComponent<FakeDequer>(DependencyLifecycle.SingleInstance));
+            config.RegisterComponents(c => c.ConfigureComponent<FakeSender>(DependencyLifecycle.SingleInstance));
+            config.RegisterComponents(c => c.ConfigureComponent<FakeQueueCreator>(DependencyLifecycle.SingleInstance));
+        }
+    }
+
+    class FakeDequer : IDequeueMessages
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Timeouts/TransportWithFakeQueues.cs]: shouldn't that be FakeDequeuer?

File: src/NServiceBus.AcceptanceTests/Timeouts/TransportWithFakeQueues.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {
+
+        public void Init(Address address, Unicast.Transport.TransactionSettings transactionSettings, Func<TransportMessage, bool> tryProcessMessage, Action<TransportMessage, Exception> endProcessMessage)
+        {
+        }
+
+        public void Start(int maximumConcurrencyLevel)
+        {
+
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Timeouts/TransportWithFakeQueues.cs]: let's remove the empty lines

**Comment by timbussmann** [comment]
@andreasohlund can you review the v5 fix?

**Comment by weralabaj** [comment]
@Particular/nservicebus Please, review.

File: src/NServiceBus.AcceptanceTests/Exceptions/When_Uow_Begin_throws.cs [hunk 3]
```diff
@@ -1,123 +0,0 @@
-﻿namespace NServiceBus.AcceptanceTests.Exceptions
```
Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Exceptions/When_Uow_Begin_throws.cs]: Remind me why we decided to kill those tests?`( I know there was a reason)

File: src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        /// <param name="configure"></param>
+        /// <returns></returns>
+        public static BusConfiguration SuppressOutdatedTimeoutPersistenceWarning(this BusConfiguration configure)
+        {
+            configure.Settings.Set(TimeoutPersistenceVersionCheck.SuppressOutdatedTimeoutPersistenceWarning, true);
+            return configure;
+        }
+    }
+
+    internal class TimeoutPersistenceVersionCheck : IWantToRunWhenConfigurationIsComplete
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs]: Redunant internal + move to a seprate file

File: src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+{
+    using NServiceBus.Config;
+    using System;
+    using System.Configuration;
+    using NServiceBus.ObjectBuilder;
+    using NServiceBus.Settings;
+    using NServiceBus.Transports;
+
+    /// <summary>
+    ///
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs]: Empty api docs

File: src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    using NServiceBus.Settings;
+    using NServiceBus.Transports;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public static class TimeoutPersistenceVersionCheckExtension
+    {
+        /// <summary>
+        ///
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs]: Empty api docs

File: src/NServiceBus.Core/Timeout/Hosting/Windows/TimeoutDispatcherProcessor.cs [hunk 7]
```diff
@@ -19,20 +21,47 @@ public TimeoutDispatcherProcessor()
...
+                if (timeoutData == null)
+                {
+                    return true;
+                }
+
+                var sendOptions = timeoutData.ToSendOptions(Configure.LocalAddress);
+                
+                if (ShouldSuppressTransaction())
+                {
+                    sendOptions.EnlistInReceiveTransaction = false;
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Timeout/Hosting/Windows/TimeoutDispatcherProcessor.cs]: Have we double checked that the v5 versions of MSMQ and SqlServer honors this?

File: src/NServiceBus.AcceptanceTests/Exceptions/When_Uow_Begin_throws.cs (hunk 3 above)
Comment (SimonCropp) [inline src/NServiceBus.AcceptanceTests/Exceptions/When_Uow_Begin_throws.cs]: I gave in. Tests are too fragile and dont add enough value

Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Exceptions/When_Uow_Begin_throws.cs]: :+1:

File: src/NServiceBus.Core/Timeout/Hosting/Windows/TimeoutDispatcherProcessor.cs (hunk 7 above)
Comment (timbussmann) [inline src/NServiceBus.Core/Timeout/Hosting/Windows/TimeoutDispatcherProcessor.cs]: I checked both, MSMQ only suppresses native transactions while SQL suppresses all transactions as far as I understood the code. But probably better if a second pair of eyes verifies this.

**Comment by timbussmann** [comment]
switched to PR #2963

**Pull request closed without merging**